### ISSUE TITLE
BUG: with 2.5.2 tests fail on arm64, armhf, mips64el & mipsel

### ISSUE THREAD

**Issue by josch** [seccomp/libseccomp#338]
Hi,

version 2.5.2 was uploaded to Debian and the tests fail on several architectures:

https://buildd.debian.org/status/package.php?p=libseccomp

For detailed build logs, click on the red "Build-Attempted" table cells.

I bisected libseccomp from git on two of the failing architectures and figured out that the failure on arm64 was introduced in aa0f858aa58d51c93a176c60a4c83a4a303bcffd and the failure on mips exists since e976080ac5547a46f4654abc710e89bed992bb63.

**Comment by drakenclimber** [comment]
Thanks for the submission, @josch.  The arm64 failure is likely similar to Issue https://github.com/seccomp/libseccomp/issues/336.  As currently written, test 11 will fail on older kernels that don't support the `SECCOMP_GET_NOTIF_SIZES` flag.

Failures on other architectures has been a fear of mine since we lost our automated testing on arm64, ppc, etc.  I wish I had a better answer :/

**Comment by debfx** [comment]
One of the tests that fails on mipsel is:

```
	rc = seccomp_rule_add_exact(ctx, SCMP_ACT_KILL, SCMP_SYS(socket), 1,
				    SCMP_A0(SCMP_CMP_EQ, 2));
	if (rc != -EINVAL)
		return -1;
```

(from 11-basic-basic_errors.c)

**Comment by pcmoore** [comment]
Hi @josch, I'm not sure this will solve all of your problems, but could you try PR #346?

**Comment by pcmoore** [comment]
Thanks @debfx and @josch.  Looking at the log that @debfx posted I'm most concerned with the "02-sim-basic" test failures as that is a pretty basic test and the failures would indicate that something fundamental has gone wrong.  This is especially perplexing as most of the arch specific syscall lookup code is now common, see f454456e261930d94b3a1a444b6bac75c11c3cb0.

@debfx and @josch do either of you have easy~ish access to a MIPS system?  Would you be willing to attempt some basic debugging for us?  I suspect if we resolve the "02-sim-basic" problems we should resolve a lot of the other failures as well.

**Comment by pcmoore** [comment]
Bummer.  @debfx can you run the test program directly (see below) and put the result somewhere?  It is probably a bit too large for this issue, but perhaps a linked GH gist?
```
% make check-build
% cd tests
% ./05-sim-long_jumps
```

**Comment by debfx** [comment]
It doesn't output anything.

The call in line 63 returns -EINVAL on iter=168:

```c
		rc = seccomp_rule_add_exact(ctx, SCMP_ACT_ALLOW, iter, 1,
					    SCMP_A0(SCMP_CMP_NE, 0));
		if (rc != 0)
			goto out;
```

When I remove the goto, I can see that with iter 168 - 184 it returns -EINVAL. iter 185 - 199 return 0 again.

**Comment by pcmoore** [comment]
Closing this out as I believe we have fixed this in #348, feel free to reopen if the problem remains.
